src/01-general/work-orders.apib

```diff
@@ -0,0 +1,199 @@
+
+### workOrders.attachImage [POST /workOrders.attachImage]
+
+Attach image to a draft work order.
+
++ Request (application/json;charset=utf-8)
+
+    + Attributes (object)
+        + id: `4afb0a9c-91c6-49ed-a2e5-ce7c1e3a87fb` (string, required)
+        + image_data: `data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAIAAACQkWg2AAAAAXNSR0IArs4c6QAAAVlpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8AAAASUVORK5CYII=` (string, required) - PNG image, base64 encoded
+
++ Response 200 (application/json;charset=utf-8)
+
+    + Attributes (object)
+        + data (object)
+            + type: `workOrderImage` (string)
+            + id: `a856642a-a476-4b69-8e42-e7d1badb2ce9` (string)
+
+### workOrders.listImages [GET /workOrders.listImages]
+
+List images attached to a work order.
+
++ Request (application/json;charset=utf-8)
+
+    + Attributes (object)
+        + id: `4afb0a9c-91c6-49ed-a2e5-ce7c1e3a87fb` (string, required)
+
++ Response 200 (application/json;charset=utf-8)
+
+    + Attributes (object)
+        + data (array)
+            + (object)
+                + id: `a856642a-a476-4b69-8e42-e7d1badb2ce9` (string)
+                + image_data: `data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAIAAACQkWg2AAAAAXNSR0IArs4c6QAAAVlpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8AAAASUVORK5CYII=` (string) - PNG image, base64 encoded
+
+### workOrders.deleteImage [POST /workOrders.deleteImage]
+
+Delete image from a draft work order.
+
++ Request (application/json;charset=utf-8)
+
+    + Attributes (object)
+        + id: `4afb0a9c-91c6-49ed-a2e5-ce7c1e3a87fb` (string, required)
+        + image_id: `a856642a-a476-4b69-8e42-e7d1badb2ce9` (string, required)
+
++ Response 204
+
+### workOrders.sign [POST /workOrders.sign]
+
+Sign a draft work order.
+
++ Request (application/json;charset=utf-8)
+
+    + Attributes (object)
+        + id: `4afb0a9c-91c6-49ed-a2e5-ce7c1e3a87fb` (string, required)
+        + signature_data: `data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAIAAACQkWg2AAAAAXNSR0IArs4c6QAAAVlpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8AAAASUVORK5CYII=` (string, required) - PNG image, base64 encoded
+
++ Response 204
+
+### workOrders.delete [POST /workOrders.delete]
+
+Delete a draft work order.
+
++ Request (application/json;charset=utf-8)
+
+    + Attributes (object)
+        + id: `4afb0a9c-91c6-49ed-a2e5-ce7c1e3a87fb` (string, required)
+
++ Response 204
+
+### workOrders.finalize [POST /workOrders.finalize]
+
+Finalize a draft work order by generating the PDF.
+
++ Request (application/json;charset=utf-8)
+
+    + Attributes (object)
+        + id: `4afb0a9c-91c6-49ed-a2e5-ce7c1e3a87fb` (string, required)
+
++ Response 204
+
+### workOrders.send [POST /workOrders.send]
+
+Send a finalized work order to the customer.
+
++ Request (application/json;charset=utf-8)
+
+    + Attributes (object)
```

Comment:
Q : maybe we have to provide sending details here too? who to send it to, where to send duplicates, which layout to use, ...

Reply:
Additional properties will become obvious when we groom the relevant stories. Going to leave it like this for now.